scipy/linalg/tests/test_decomp.py

```diff
@@ -913,144 +903,6 @@ def test_deprecation_results(self):
         assert_allclose(v_dep, v)
 
 
-class TestLU:
-    def setup_method(self):
-        self.a = array([[1, 2, 3], [1, 2, 3], [2, 5, 6]])
-        self.ca = array([[1, 2, 3], [1, 2, 3], [2, 5j, 6]])
-        # Those matrices are more robust to detect problems in permutation
-        # matrices than the ones above
-        self.b = array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
-        self.cb = array([[1j, 2j, 3j], [4j, 5j, 6j], [7j, 8j, 9j]])
-
-        # Reectangular matrices
-        self.hrect = array([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 12, 12]])
-        self.chrect = 1.j * array([[1, 2, 3, 4],
-                                   [5, 6, 7, 8],
-                                   [9, 10, 12, 12]])
-
-        self.vrect = array([[1, 2, 3], [4, 5, 6], [7, 8, 9], [10, 12, 12]])
-        self.cvrect = 1.j * array([[1, 2, 3],
-                                   [4, 5, 6],
-                                   [7, 8, 9],
-                                   [10, 12, 12]])
-
-        # Medium sizes matrices
-        self.med = random((30, 40))
-        self.cmed = random((30, 40)) + 1.j * random((30, 40))
-
-    def _test_common(self, data):
```

Comment:
It's not obvious that this was replaced (e.g. there is no longer a method with this name). Is this no longer needed, or was it replaced with the other new tests to `TestLU`? (Does it have to be replaced? It's easier to believe that things are safe if tests are not rewritten.)

Reply:
All tests were combined for `lu`, `lu_factor`, and `lu_solve` and `test_common` was testing for `lu`. I took out the LU parts out and left the remaining ones in to not to cause any issues with those ones. At some point we can refactor those ones too then they will become the legacy tests.

Reply:
I'm not sure I understand.

> All tests were combined for `lu`, `lu_factor`, and `lu_solve`

`lu_solve` tests were separate from the `lu` and `lu_factor` tests before, and that has not changed.

>  I took out the LU parts out and left the remaining ones in to not to cause any issues with those ones.

Previously, `lu` and `lu_factor` were both tested in `TestLU`. By "left the remaining ones in", do you mean that you moved all the `lu_factor` tests into a new class `TestLUFactor`, and rewrote all the `lu` tests?

---

I'll try rephrasing my question:
Have all the `lu` tests been rewritten? If so, why? 
This test makes major changes to `lu`. If we want to ensure backward compatibility, we should *at least* make sure that all the old tests pass. Why not leave the old test alone, and add new ones as needed?

Reply:
The old tests were, as you can follow from `lu_factor`, rather trivial small examples. So I tested with them anyways but they are far from testing anything substantial. The new tests spanning more sizes and cases. Old ones are just going to make noise with lots of manual array entries.

Reply:
As you wish, since I won't be merging. My hesitation is that I don't want to review the old tests to see that they are strictly being improved. So I suppose I will just run them locally to be sure that they still pass and otherwise ignore them, reviewing the new tests from scratch.

Reply:
Sure no problem at all,